packages/multi-test/src/app.rs

```diff
@@ -776,4 +776,45 @@ mod test {
         let committed = query_app(&app, &rcpt);
         assert_eq!(coins(37, "eth"), committed);
     }
+
+    #[test]
+    fn sent_funds_properly_visible_on_execution() {
+        // Testing if funds on contract are properly visible on contract.
+        // Hackatom contract is initialized with 10btc. Then, the contract is executed, with
+        // additional 20btc. Then beneficiary balance is checked - expeced value is 30btc. 10btc
+        // would mean that sending tokens with message is not visible for this very message, and
+        // 20btc means, that only such just send funds are visible.
+        let mut app = mock_app();
+
+        let owner = Addr::unchecked("owner");
+        let beneficiary = Addr::unchecked("beneficiary");
+        app.init_bank_balance(&owner, coins(30, "btc")).unwrap();
+
+        let contract_id = app.store_code(hackatom::contract());
+        let contract = app
+            .instantiate_contract(
+                contract_id,
+                owner.clone(),
+                &hackatom::InitMsg {
+                    beneficiary: beneficiary.as_str().to_owned(),
+                },
+                &coins(10, "btc"),
+                "Hackatom",
+            )
+            .unwrap();
+
+        app.execute_contract(
+            owner.clone(),
+            contract.clone(),
+            &EmptyMsg {},
+            &coins(20, "btc"),
+        )
+        .unwrap();
+
+        // Check balance of all accounts to ensure no tokens where burned or created, and they are
+        // in correct places
+        assert_eq!(get_balance(&app, &owner), &[]);
```

Comment:
Awesome work.

I would usually add such asserts at the state between instantiate and execute, but you are correct, those are totally not needed for such a BDD test like this. If these failed for some reason, I would add some intermediary checks, otherwise it is just noise.

Looks good to me.

Reply:
I just try to find a balance between checking everything, and deciding what is relevant to case. Basically it is nice when you can understand the test just looking at it quickly. It would be ideal to test everything (like even to check if some side state didn't change - which there is no reason to happen, but for sake of completeness), but I found this checks should actually cover test case.